inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -703,11 +702,71 @@ Parameter clDNNEngine::GetMetric(const std::string& name, const std::map<std::st
     } else if (name == METRIC_KEY(RANGE_FOR_STREAMS)) {
         std::tuple<unsigned int, unsigned int> range = std::make_tuple(1, 2);
         IE_SET_METRIC_RETURN(RANGE_FOR_STREAMS, range);
+    } else if (name == GPU_METRIC_KEY(MAX_BATCH_SIZE)) {
+        const auto& config = _impl->m_configs.GetConfig(device_id);
+        auto n_streams = config.throughput_streams;
+        auto available_device_mem = device_info.max_global_mem_size;
+        int64_t max_batch_size = 0;
+
+        if (options.find("CNN_NETWORK") == options.end()) {
+            throw std::runtime_error("No CNN_NETWORK option given!");
+        }
+        if (options.find("GPU_THROUGHPUT_STREAMS") != options.end()) {
+            n_streams = options.find("GPU_THROUGHPUT_STREAMS")->second.as<uint16_t>();
+        }
+        if (options.find("AVAILABLE_DEVICE_MEM_SIZE") != options.end()) {
+            available_device_mem = options.find("AVAILABLE_DEVICE_MEM_SIZE")->second.as<int64_t>();
+            if (available_device_mem < 0)
+                throw std::runtime_error("No available device mem");
```

Comment:
`IE_THROW() << "[GPU] AVAILABLE_DEVICE_MEM_SIZE value should be greater than 0 for max batch size calculation"`

Reply:
Applied